agreement/player.go

```diff
@@ -158,8 +158,16 @@ func (p *player) handleFastTimeout(r routerHandle, e timeoutEvent) []action {
 	return p.issueFastVote(r)
 }
 
-func (p *player) issueSoftVote(r routerHandle) (actions []action) {
+func (p *player) issueSoftVote(r routerHandle, te timeoutEvent) (actions []action) {
 	defer func() {
+		var deadlineTimeout time.Duration
+		if te.Proto.Version == "" || te.Proto.Err != nil {
+			r.t.log.Errorf("failed to read valid protocol version for timeout event (proto %v): %v. "+
+				"Falling Back to default deadline timeout.", te.Proto.Version, te.Proto.Err)
+			deadlineTimeout = defaultDeadlineTimeout
```

Comment:
It seems a bit undesirable to shadow the global `deadlineTimeout` variable name.  Might be better to pick something that's unambiguous.

Reply:
Ah, never mind, didn't see the other change in the PR.  This is fine.